### PR TITLE
librealsense@1 1.12.1 (new formula)

### PR DESCRIPTION
**Pull request by OrganicIrradiation**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/librealsense@1.12.1.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class LibrealsenseAT1121 < Formula
```
Comment (ilovezfs) [inline Formula/librealsense@1.12.1.rb]: AT1

File: Formula/librealsense@1.12.1.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  desc "Intel RealSense F200, SR300, R200, LR200 and ZR300 capture"
+  homepage "https://github.com/IntelRealSense/librealsense/tree/legacy"
+  url "https://github.com/IntelRealSense/librealsense/archive/v1.12.1.tar.gz"
+  sha256 "62fb4afac289ad7e25c81b6be584ee275f3d4d3742468dc7d80222ee2e4671bd"
+  head "https://github.com/IntelRealSense/librealsense.git", :branch => "legacy"
+
+  option "with-examples", "Install examples"
```
Comment (ilovezfs) [inline Formula/librealsense@1.12.1.rb]: don't need this

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Is upstream maintaining this as a separate release line? I don't see any evidence of that but I might be missing something.

**Comment by OrganicIrradiation** [comment]
The main repo refers any owners of the F200, R200, LR200 and ZR300 devices to the [last legacy release](https://github.com/IntelRealSense/librealsense/tree/v1.12.1) (see second sentence here: https://github.com/IntelRealSense/librealsense), which was released Dec. 26, 2016. However, this legacy branch is being actively maintained, last being updated Oct 3, 2017. The bump to the 2.x branch by @ilovezfs in baf5d33b2f32effc216eaae58f61cff18fdcd9bb broke a lot of support for legacy devices.

File: Formula/librealsense@1.12.1.rb (hunk 1 above)
Comment (OrganicIrradiation) [reply Formula/librealsense@1.12.1.rb]: To clarify, then the filename should be changed to librealsense@1 as well?

File: Formula/librealsense@1.12.1.rb (hunk 2 above)
Comment (OrganicIrradiation) [reply Formula/librealsense@1.12.1.rb]: The examples flag was present before the bump to the 2.x branch and can add (unnecessary) compile time if default on: baf5d33b2f32effc216eaae58f61cff18fdcd9bb

If you're suggesting the examples are not necessary, then I'd disagree. They are useful for debugging the devices.

File: Formula/librealsense@1.12.1.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/librealsense@1.12.1.rb]: yes

File: Formula/librealsense@1.12.1.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/librealsense@1.12.1.rb]: Either we include them always or exclude them always (I'll leave it up to you). I don't want options on a new versioned formula.

File: Formula/librealsense@1.12.1.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      #include <librealsense/rs.h>
+      #include<stdio.h>
+      int main()
+      {
+        printf(RS_API_VERSION_STR);
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-I#{include}", "-L#{lib}", "-o", "test"
+    assert_equal shell_output("./test").strip, version.to_s
```
Comment (ilovezfs) [inline Formula/librealsense@1.12.1.rb]: The order of argument is reversed here. It should be expected, actual not actual, expected

File: Formula/librealsense@1.12.1.rb (hunk 2 above)
Comment (OrganicIrradiation) [reply Formula/librealsense@1.12.1.rb]: Ok, sounds fine, I'll include them.

**Review by OrganicIrradiation — COMMENTED** [review]
All requested changes addressed in 85bb343eb8cd8721fd4a48ddb0f1f5fb57cbc19a

**Comment by ilovezfs** [comment]
@OrganicIrradiation it only failed because you didn't remove the `if build.with? "examples"` from the glfw line.

File: Formula/librealsense@1.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  sha256 "62fb4afac289ad7e25c81b6be584ee275f3d4d3742468dc7d80222ee2e4671bd"
+  head "https://github.com/IntelRealSense/librealsense.git", :branch => "legacy"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glfw"
+  depends_on "libusb"
+
+  def install
+    args = std_cmake_args
```
Comment (ilovezfs) [inline Formula/librealsense@1.rb]: just inline these below

```
    system "cmake", "-DBUILD_EXAMPLES=true", *std_cmake_args
```

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/librealsense@1.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    args << "-DBUILD_EXAMPLES=true"
+
+    system "cmake", ".", *args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <librealsense/rs.h>
+      #include<stdio.h>
```
Comment (ilovezfs) [inline Formula/librealsense@1.rb]: add a space please

File: Formula/librealsense@1.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 "62fb4afac289ad7e25c81b6be584ee275f3d4d3742468dc7d80222ee2e4671bd"
+  head "https://github.com/IntelRealSense/librealsense.git", :branch => "legacy"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glfw"
+  depends_on "libusb"
+
+  def install
+    system "cmake", "-DBUILD_EXAMPLES=true", *std_cmake_args
```
Comment (ilovezfs) [inline Formula/librealsense@1.rb]: does `=ON` work here? That is usually what we have

Comment (OrganicIrradiation) [reply Formula/librealsense@1.rb]: Yep, just tested and updated.

**Comment by ilovezfs** [comment]
Do the files installed by this overlap with any of the files installed by @2? It's likely we need `keg_only :versioned_formula`

**Comment by OrganicIrradiation** [comment]
I don't believe so:

```
$ brew install --verbose librealsense@1
...
/usr/local/Cellar/cmake/3.10.1/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/lib/librealsense.1.12.1.dylib
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/lib/librealsense.1.dylib
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/lib/librealsense.dylib
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/include/librealsense
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/include/librealsense/rscore.hpp
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/include/librealsense/rs.hpp
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/include/librealsense/rs.h
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/include/librealsense/rsutil.h
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/c-tutorial-1-depth
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/c-tutorial-2-streams
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/c-tutorial-3-pointcloud
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-1-depth
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-2-streams
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-3-pointcloud
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-alignimages
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-callback
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-callback-2
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-capture
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-config-ui
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-enumerate-devices
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-headless
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-motion-module
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-multicam
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-pointcloud
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-restart
-- Installing: /usr/local/Cellar/librealsense@1/1.12.1/bin/cpp-stride
==> Cleaning
==> Finishing up
ln -s ../Cellar/librealsense@1/1.12.1/bin/c-tutorial-1-depth c-tutorial-1-depth
ln -s ../Cellar/librealsense@1/1.12.1/bin/c-tutorial-2-streams c-tutorial-2-streams
ln -s ../Cellar/librealsense@1/1.12.1/bin/c-tutorial-3-pointcloud c-tutorial-3-pointcloud
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-alignimages cpp-alignimages
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-callback cpp-callback
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-callback-2 cpp-callback-2
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-capture cpp-capture
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-config-ui cpp-config-ui
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-enumerate-devices cpp-enumerate-devices
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-headless cpp-headless
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-motion-module cpp-motion-module
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-multicam cpp-multicam
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-pointcloud cpp-pointcloud
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-restart cpp-restart
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-stride cpp-stride
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-1-depth cpp-tutorial-1-depth
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-2-streams cpp-tutorial-2-streams
ln -s ../Cellar/librealsense@1/1.12.1/bin/cpp-tutorial-3-pointcloud cpp-tutorial-3-pointcloud
ln -s ../Cellar/librealsense@1/1.12.1/include/librealsense librealsense
ln -s ../Cellar/librealsense@1/1.12.1/lib/librealsense.1.12.1.dylib librealsense.1.12.1.dylib
ln -s ../Cellar/librealsense@1/1.12.1/lib/librealsense.1.dylib librealsense.1.dylib
ln -s ../Cellar/librealsense@1/1.12.1/lib/librealsense.dylib librealsense.dylib
```

```
$ brew install --verbose librealsense@2
...
/usr/local/Cellar/cmake/3.10.1/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/librealsense2.2.9.0.dylib
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/librealsense2.2.dylib
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/librealsense2.dylib
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/rs_advanced_mode.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/rs.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_device.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_internal.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_processing.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_record_playback.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_pipeline.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_types.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_frame.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_context.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/hpp/rs_sensor.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/rs.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_frame.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_context.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_record_playback.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_device.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_sensor.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_types.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_pipeline.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_internal.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_processing.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_advanced_mode_command.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/h/rs_option.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/rsutil.h
-- Installing: /usr/local/Cellar/librealsense/2.9.0/include/librealsense2/rs_advanced_mode.hpp
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/cmake/realsense2/realsense2Targets.cmake
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/cmake/realsense2/realsense2Targets-release.cmake
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/cmake/realsense2/realsense2Config.cmake
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/pkgconfig/realsense2.pc
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/libusb.a
-- Installing: /usr/local/Cellar/librealsense/2.9.0/lib/librealsense-file.a
-- Installing: /usr/local/Cellar/librealsense/2.9.0/bin/live-test
==> Cleaning
==> Finishing up
ln -s ../Cellar/librealsense/2.9.0/bin/live-test live-test
ln -s ../Cellar/librealsense/2.9.0/include/librealsense2 librealsense2
ln -s ../../Cellar/librealsense/2.9.0/lib/cmake/realsense2 realsense2
ln -s ../Cellar/librealsense/2.9.0/lib/librealsense-file.a librealsense-file.a
ln -s ../Cellar/librealsense/2.9.0/lib/librealsense2.2.9.0.dylib librealsense2.2.9.0.dylib
ln -s ../Cellar/librealsense/2.9.0/lib/librealsense2.2.dylib librealsense2.2.dylib
ln -s ../Cellar/librealsense/2.9.0/lib/librealsense2.dylib librealsense2.dylib
ln -s ../Cellar/librealsense/2.9.0/lib/libusb.a libusb.a
ln -s ../../Cellar/librealsense/2.9.0/lib/pkgconfig/realsense2.pc realsense2.pc
```

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @OrganicIrradiation! :rocket:

**Comment by OrganicIrradiation** [comment]
It looks like @fxcoudert removed this formula. Any reason behind the removal? I can no longer access my legacy RealSense camera via Homebrew, although the 1.x branch is still being separately maintained.

**Comment by OrganicIrradiation** [comment]
I see it was removed because it is not often used: https://github.com/Homebrew/homebrew-core/pull/23083

I'm disappointed by this decision.

**Comment by fxcoudert** [comment]
We have recently tightened our criteria for support of versioned formulas: https://github.com/Homebrew/brew/pull/3697

`librealsense@1` was seeing about 4 installs per month, which indicates it's not widely used at all, and may be a better fit for a personal tap: https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md

**Pull request closed without merging**